slashing-protection/src/integration-test/java/tech/pegasys/web3signer/slashingprotection/PruningIntegrationTest.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2021 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.web3signer.slashingprotection;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import tech.pegasys.web3signer.slashingprotection.dao.SignedAttestation;
+import tech.pegasys.web3signer.slashingprotection.dao.SignedBlock;
+import tech.pegasys.web3signer.slashingprotection.dao.SigningWatermark;
+
+import java.util.List;
+
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.units.bigints.UInt64;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+public class PruningIntegrationTest extends IntegrationTestBase {
+
+  @ParameterizedTest
+  @CsvSource({"1, 9", "2, 8", "5, 5", "9, 1", "10, 0", "20, 0"})
+  void prunesDataForRegisteredValidator(final int amountToKeep, final int pruningMark) {
```

Comment:
do you need both numbers as parameters - it appears 90% of the time they add to 10 (which is size) - which makes me think we could write this as {entiresInDataBase, numberTo prune} ... ?

Reply:
renamed argument to expectedLowestPopulatedSlot, hopefully this is clearer